cli/tests/test_fix_command.rs

```diff
@@ -0,0 +1,513 @@
+    // in the root directory of the repo, which is worth reconsidering if we
+    // establish a contract regarding cwd.
+    let (test_env, repo_path) = init_with_fake_formatter("--reverse --tee {path}-fixlog");
+
+    // There are at least two interesting cases: the content is repeated immediately
+    // in the child commit, or later in another descendant.
+    std::fs::write(repo_path.join("file"), "foo\n").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["branch", "create", "a"]);
+    test_env.jj_cmd_ok(&repo_path, &["new"]);
+    std::fs::write(repo_path.join("file"), "bar\n").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["branch", "create", "b"]);
+    test_env.jj_cmd_ok(&repo_path, &["new"]);
+    std::fs::write(repo_path.join("file"), "bar\n").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["branch", "create", "c"]);
+    test_env.jj_cmd_ok(&repo_path, &["new"]);
+    std::fs::write(repo_path.join("file"), "foo\n").unwrap();
+    test_env.jj_cmd_ok(&repo_path, &["branch", "create", "d"]);
+
+    let (stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["fix", "-s", "a"]);
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(stderr, @r###"
+    Fixed 4 commits of 4 checked.
+    Working copy now at: yqosqzyt bcb48084 d | (no description set)
+    Parent commit      : mzvwutvl c09b05f5 c | (empty) (no description set)
+    Added 0 files, modified 1 files, removed 0 files
+    "###);
+    let content = test_env.jj_cmd_success(&repo_path, &["print", "file", "-r", "a"]);
+    insta::assert_snapshot!(content, @"oof\n");
+    let content = test_env.jj_cmd_success(&repo_path, &["print", "file", "-r", "b"]);
+    insta::assert_snapshot!(content, @"rab\n");
+    let content = test_env.jj_cmd_success(&repo_path, &["print", "file", "-r", "c"]);
+    insta::assert_snapshot!(content, @"rab\n");
+    let content = test_env.jj_cmd_success(&repo_path, &["print", "file", "-r", "d"]);
+    insta::assert_snapshot!(content, @"oof\n");
+
+    // Each new content string only appears once in the log, because all the other
+    // inputs (like file name) were identical, and so the results were re-used. We
+    // sort the log because the order of execution inside `jj fix` is undefined.
+    insta::assert_snapshot!(sorted_lines(repo_path.join("file-fixlog")), @"oof\nrab\n");
+}
+
+fn sorted_lines(path: PathBuf) -> String {
+    let mut log: Vec<_> = std::fs::read_to_string(path.as_os_str())
+        .unwrap()
+        .lines()
+        .map(String::from)
+        .collect();
+    log.sort();
+    log.join("\n")
+}
+
+#[test]
+fn test_no_op() {
+    let (test_env, repo_path) = init_with_fake_formatter("");
+    std::fs::write(repo_path.join("file"), "content").unwrap();
+
+    let (stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["fix", "-s", "@"]);
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(stderr, @r###"
+    Fixed 1 commits of 1 checked.
```

Comment:
What was fixed here? The content looks the same before and after, and the test case is called "no_op"...